db/migrate/20220625142729_convert_map_to_note_nodes.rb

```diff
@@ -0,0 +1,76 @@
+class ConvertMapToNoteNodes < ActiveRecord::Migration[5.2]
+  def change
+    Node.where(type: 'map').each do |node|
+      revision = node.latest
+      iframe_link = "<iframe style='width:100%;border:none;background:#ddd;margin-bottom:8px;' height='375' src='https://publiclab.github.io/plots-leaflet-viewer/?tms=#{node.map.tms}&lon=#{node.lon}&lat=#{node.lat}&zoom=#{node.map.min_zoom+1}'></iframe>"
+      conversion_buttons =
+        "<div class='btn-toolbar'>"\
+          "<div class='btn-group'>"\
+            "<a class='btn btn-sm btn-danger' rel='tooltip' title='View the map in a full-screen web viewer' href='https://publiclab.github.io/plots-leaflet-viewer/?tms=#{node.map.tms}&lon=#{node.lon}&lat=#{node.lat}&zoom=17'>Web viewer</a>"\
+            "<a class='btn btn-sm btn-success' rel='tooltip' title='' href='#{node.map.field_jpg_url_value}'>JPG (#{node.map.field_jpg_filesize_value.to_i} MB)</a>"\
+            "<a class='btn btn-sm btn-info' rel='tooltip' title='' href='#{node.map.field_geotiff_url_value}'>GeoTiff (#{node.map.field_geotiff_filesize_value.to_i} MB)</a>"\
```

Comment:
We might need to think on how to handle instances where some of the values here are nil nil e.g `field_geotiff_filesize_value` and  `field_geotiff_filesize_value`. Maybe if we can get unstable running we can see how these links look like.  I ran this for one map locally and some of this were nil this  is what I get 
```
"<a class='btn btn-sm btn-danger' rel='tooltip' title='View the map in a full-screen web viewer' href='https://publiclab.github.io/plots-leaflet-viewer/?tms=https://archive.publiclaboratory.org/leaflet/?    tms=https://archive.publiclab.org/2013/2013-04-15-us-massachusetts-plum-    island/tms/&lon=-70.80848&lat=42.7952&zoom=16&lon=23.0&lat=70.0&zoom=17'>Web viewer</a><a class='btn btn-sm btn-success' rel='tooltip' title='' href=''>JPG (0 MB)</a><a class='btn btn-sm btn-info' rel='tooltip' title='' href=''>GeoTiff (0 MB)</a>"
```
Everything else is looking great . Good job!

Reply:
Hmm yes. When the values are nil, running `.to_i` on it returns `0` I believe. That's why the size for the JPG and GeoTiff is displayed as 0MB

Reply:
Sounds good - yes, perhaps we add a conditional to avoid adding them, OR we add a "disabled" class version, and/or a tooltip saying it's not available?

Reply:
I've added a "disabled" class to the button when the value is `nil` (i.e. 0MB)